api/envoy/extensions/filters/http/mcp/v3/mcp.proto

```diff
@@ -54,6 +55,12 @@ message Mcp {
 
   // Parser configuration, this provide the attribute extraction override.
   ParserConfig parser_config = 4;
+
+  // When set, enables validation that the subject embedded in the session ID
+  // matches the authenticated identity from the request.
+  // This prevents session hijacking by ensuring the session belongs to the authenticated user.
+  // If validation fails, the request is rejected with 403 Forbidden.
+  SubjectValidation subject_validation = 5;
```

Comment:
I started to think we can move this config to the mcp_router, since the session id is reconstructed only by the mcp_router. The MCP filter can be used standalone without the router so the session_id can be any customized stuffs.

The logic in the router now is if there is no session id, we re-construct a new id, and if there is an id, we check the new id with the subject.